sample/src/main/java/com/canhub/cropper/sample/camera_java/presenter/SCameraPresenterJava.java

```diff
@@ -0,0 +1,169 @@
+package com.canhub.cropper.sample.camera_java.presenter;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Build;
+import android.util.Log;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import com.canhub.cropper.CropImage;
+import com.canhub.cropper.sample.camera_java.app.SCameraFragmentJava;
+import com.canhub.cropper.sample.camera_java.domain.CameraEnumDomainJava;
+import com.canhub.cropper.sample.camera_java.domain.SCameraContractJava;
+import com.canhub.cropper.sample.camera_java.app.SCameraFragmentJava;
+
+import static android.app.Activity.RESULT_OK;
+import static com.canhub.cropper.sample.camera_java.app.SCameraFragmentJava.CODE_PHOTO_CAMERA;
+
+public class SCameraPresenterJava implements SCameraContractJava.Presenter {
+    private SCameraContractJava.View view = null;
+    private boolean minVersion = android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
+    private boolean request = false;
+    private boolean hasSystemFeature = false;
+    private boolean selfPermission = false;
+    private Context context = null;
+
+    @Override
+    public void bind(SCameraContractJava.View view) {
+        this.view = view;
+    }
+
+    @Override
+    public void unbind() {
+        view = null;
+    }
+
+    @Override
+    public void onPermissionResult(boolean granted) {
+        assert view!=null;
+        if(granted){
+            view.dispatchTakePictureIntent();
+        } else if(minVersion && request){
+            view.showDialog();
+        } else {
+            view.cameraPermissionLaunch();
+        }
+    }
+
+    @Override
+    public void onCreate(FragmentActivity activity, Context context) {
+        assert view!=null;
+        if(activity==null || context==null){
+            view.showErrorMessage("onCreate activity and/or context are null");
+            return;
+        }
+        this.context = context;
+
+        request = ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.CAMERA);
+        assert context.getPackageManager()!=null;
+        hasSystemFeature = context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY);
```

Comment:
This is Nullable in Java. Please add a check

`if(hasSystemFeature == null) hasSystemFeature = false`

Or something like this

Reply:
hasSystemFeature is **boolean** that in java is true or false.
Should I change it to **Boolean** so I can give null checker?
But I think more efficient in **boolean**

Reply:
What happen in this code if `context.getPackageManager()` returns null? 

Sadly it can return null in some cases =/, so we should treat it

Reply:
ah sorry, I notice that my assert won't do that. I already commit the fixes